File: Iteration2/code/CSE3063GRP12/src/features/advisor/student_list/AdvisorStudentListController.java
package features.advisor.student_list;

import core.general_providers.TerminalManager;
import core.general_providers.SessionController;
import core.models.concretes.Advisor;
import core.repositories.UserRepository;
import features.main_menu.MenuController;

public class AdvisorStudentListController {
    private AdvisorStudentListView advisorStudentListView;
    private UserRepository userRepository;

    public AdvisorStudentListController() {
        advisorStudentListView = new AdvisorStudentListView();
        userRepository = new UserRepository();

        handleStudentList();
    }

    private void navigateToMenu() {
        new MenuController();
    }

    private String getUserInput() {
        String input = TerminalManager.getInstance().read();
        //TerminalManager.getInstance().dispose();
        return input;
    }

    private void handleStudentList() {
        Advisor a = (Advisor) SessionController.getInstance().getCurrentUser();
        advisorStudentListView.showStudentList(userRepository.getStudentsByAdvisor(a));
        advisorStudentListView.showQuitMessage();
        String input = getUserInput();
        if (input.equals("q")) {
            navigateToMenu();
        }    
    }
}

File: Iteration3/code/CSE3063GRP12/src/features/advisor/student_list/AdvisorStudentListController.py
from core.general_providers.SessionController import SessionController
from core.models.concretes.Advisor import Advisor
from core.repositories.UserRepository import UserRepository
from features.advisor.student_list.AdvisorStudentListView import AdvisorStudentListView

class AdvisorStudentListController:
    def __init__(self):
        self.__advisorStudentListView = AdvisorStudentListView()
        self.__userRepository = UserRepository()
        self.handle_student_list()

    def navigate_to_menu(self):
        from features.main_menu.MenuController import MenuController
        MenuController()

    def handle_student_list(self):
        advisor = SessionController.getInstance().get_current_user()
        if isinstance(advisor, Advisor):
            self.__advisorStudentListView.showStudentList(self.__userRepository.get_students_by_advisor(advisor))
            self.__advisorStudentListView.showQuitMessage()
            user_input = input()
            try:
                if user_input == "q":
                    self.navigate_to_menu()
                else:
                    self.__advisorStudentListView.showErrorMessage()
            except Exception as e:
                print("Error:", str(e))
                self.handle_student_list()
